torch/csrc/jit/mobile/debug_info.cpp

```diff
@@ -0,0 +1,51 @@
+#include <torch/csrc/jit/mobile/debug_info.h>
+#include <torch/csrc/jit/serialization/source_range_serialization.h>
+
+#include <ATen/core/ivalue.h>
+#include <torch/csrc/jit/serialization/pickle.h>
+
+#include <c10/util/string_view.h>
+
+namespace torch {
+namespace jit {
+
+MobileDebugTable::MobileDebugTable(
+    std::unique_ptr<caffe2::serialize::PyTorchStreamReader>& reader) {
+  const std::vector<std::string>& record_names = reader->getAllRecords();
+  const c10::string_view suffix("debug_pkl");
+  for (const auto& record_name : record_names) {
+    if (c10::string_view(
+          record_name.data(), record_name.size()).ends_with(suffix)) {
+      at::DataPtr debug_data;
+      size_t debug_size;
+      std::tie(debug_data, debug_size) = reader->getRecord(record_name);
+      auto ivalues =
+        jit::unpickle(
+            reinterpret_cast<const char*>(debug_data.get()), debug_size)
+        .toTuple()
+        ->elements();
+      std::unique_ptr<SourceRangeDeserializer> deserializer =
+        std::make_unique<SourceRangeDeserializer>();
+      for (auto& val : ivalues) {
+        auto tup_elems = val.toTuple()->elements();
+        TORCH_CHECK(tup_elems.size() == 3,
```

Comment:
Just wondering what's the definition of byte_offset. Is it unique? Could we reused it as the tag? Asking because adding a new field, source_tag, may break compatibility for existing models.

Reply:
Yes. Good catch. I need to check if tuples has 3 elements and then proceed. LIke in source_range_serialization.cpp. Will fix this. I will post link to quip where I documented source range details but byte offset is actually offset in the translated "code" and it is not unique for the entire model. It is unique with each code file. Thus we cannot quite use it.

Reply:
Thanks! Good to know byte offset is offset in TS the translated code. If that's the case, is it possible to use the combination of the translated code path and the offset to make it a unique key? I guess it might be used by TS to get the corresponding source range. In that sense the tag looks longer (a string instead of an integer), but would that make the schema of debug_pkl not changed?

Reply:
@iseeyuan that introduces a kind of coupling that bytecode does not need to care. For example with filename + byte offset, now you have to keep extra information in MobileDebugInfo along with tag that says which file this tag came from. That is not to say that it is not doable, but I feel it is cleaner to have separate tag compared to "filename + offset". Plus, say you want to avoid loading many archieves for reducing model load time. In that case you can potentially combine all debug_pkls into one and get rid of the "code" entirely. In that case "filename + offset" would not work. I think biggest reason to do this way is that it feels cleaner.